app/Libraries/Markdown/Osu/DocumentProcessor.php

```diff
@@ -221,6 +228,38 @@ private function recordFirstImage()
         $this->firstImage = proxy_media($this->node->getUrl());
     }
 
+    private function setImageGallery()
+    {
+        if (!$this->node instanceof Inline\Image || !$this->event->isEntering()) {
+            return;
+        }
+
+        $imageClass = $this->node->data['attributes']['class'];
+        $blockClass = $this->config->get('osu_extension/block_name');
+
+        if ($imageClass !== "{$blockClass}__figure-image") {
```

Comment:
or just do this as part of `parseFigure`. It is part of parsing figure after all.

Reply:
But `parseFigure` is done before `proxyImage`. I intentionally split this up so this is done after `proxyImage`.

Reply:
just move them around...

Reply:
So yeah, I found new problem.

`parseFigure` happen when visiting the paragraph block (the parent block), not when visiting the image block itself. So `parseRelativeUrl`, `fixWikiUrl`, and `proxyImage` not yet happen to the image block, causing the url of the image still not updated in this state.

Reply:
flip `isEntering` check so it's run last?

Reply:
Seems to work.